src/Adapter/Cart/QueryHandler/GetCartSummaryHandler.php

```diff
@@ -0,0 +1,206 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Cart\QueryHandler;
+
+use Address;
+use AddressFormat;
+use CartRule;
+use Configuration;
+use Context;
+use Customer;
+use Message;
+use PrestaShop\PrestaShop\Adapter\Entity\Cart;
+use PrestaShop\PrestaShop\Core\Domain\Cart\Query\GetCartSummary;
+use PrestaShop\PrestaShop\Core\Domain\Cart\QueryHandler\GetCartSummaryHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Cart\QueryResult\SummarizedCart;
+use Product;
+use Tools;
+
+final class GetCartSummaryHandler implements GetCartSummaryHandlerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(GetCartSummary $query)
+    {
+        $cart = new Cart($query->getCartId()->getValue());
+
+        return new SummarizedCart(
+            $this->getCartSummary($cart),
+            $this->getDeliveryOptions($cart),
+            $this->getAddresses($cart),
+            $this->isFreeShippingApplicable($cart),
+            $this->getOrderMessage($cart)
```

Comment:
i've wrapped retrieving cart summary in this query, however, in legacy controller it passed legacy `Cart` directly to ajax response like so:
```php
echo json_encode([
    'cart' => $cartObjectModel,
]);
```

i really dont want to add legacy cart to `QueryResult` object, do you have any suggestions of how to workaround this to keep BC?